### ISSUE TITLE
integration:delete command doesn't delete associated oauth consumer and oauth token

### ISSUE THREAD

**Issue by zekefarwell** [netz98/n98-magerun2#1287]
Hello and thank you for this excellent tool!  I've just run across this minor issue which is easy enough to work around, but figured it would be worth documenting anyway.

## Describe the bug

Running `n98-magerun2.phar integration:delete` removes the specified record from the `integration` table, but doesn't remove the associated records from the the `oauth_consumer` and `oauth_token` table.  After this, running `n98-magerun2.phar integration:create` with the same set of keys will throw SQL integrity constraint violations due to duplicate entries.  It might seem an odd thing to do, but this can be useful when loading a backup database into a development install that uses a different crypt key than the original install.

## Expected behaviour

Deleting an Integration from the web admin interface deletes all three records from the `integration`, `oauth_consumer` and `oauth_token` tables.  After this, running `n98-magerun2.phar integration:create` with the same set of keys works as expected to recreate the Integration. Given this, it is expected that running `n98-magerun2.phar integration:delete` would delete all the same records and a follow up `n98-magerun2.phar integration:create` with the same set of keys would work just the same.

## Steps to reproduce the issue

1. Create an integration with `n98-magerun2.phar integration:create`
2. Delete the integration with `n98-magerun2.phar integration:delete`
3. Look in the `integration` table.  See that the specified record has been deleted.
4. Look in the `oauth_consumer` and `oauth_token` tables to see that the associated records are still present.
3. Recreate the same integration with `n98-magerun2.phar integration:create` using all the same keys/tokens/secrets
4. Output should show an error like this:
```
In AbstractDb.php line 409:
                                     
  Unique constraint violation found  
                                     
In Mysql.php line 603:
                                                                                                                                                            
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '*******************************' for key 'oauth_consumer.OAUTH_CONSUMER_KEY', qu  
  ery was: UPDATE `oauth_consumer` SET `key` = ?, `secret` = ? WHERE (entity_id=4)                                                                          
                                                                                                                                                            
In Mysql.php line 109:
                                                                                                                                                            
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '*******************************' for key 'oauth_consumer.OAUTH_CONSUMER_KEY', qu  
  ery was: UPDATE `oauth_consumer` SET `key` = ?, `secret` = ? WHERE (entity_id=4)                                                                          
                                                                                                                                                            
In Mysql.php line 90:
                                                                                                                                                        
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '*******************************' for key 'oauth_consumer.OAUTH_CONSUMER_KEY'  
```

## Technical details

- Host Machine OS: Linux

## Possible Fix

I'd have expected these three tables to be linked together with an ON DELETE CASCADE constraint so that the records in the  related tables would be automatically deleted when the Integration is deleted.  That doesn't appear to be the case though.  So my guess is that all three need to be deleted separately.  I haven't looked into how the admin delete function works, but I'd imagine mirroring that in the Magerun delete command should do the trick.

**Comment by cmuench** [comment]
@zekefarwell Thanks for the detailed report 👍 
I will try to check that the next days.

**Comment by cmuench** [comment]
@zekefarwell Check it. You are right. We have to call `\Magento\Integration\Api\OauthServiceInterface::deleteConsumer` additionally.

**Comment by cmuench** [comment]
@zekefarwell fix is merged to develop branch and will be released with v7.2.0 (if ready).
You can test by updating to the develop version with `n98-magerun2.phar self-update --unstable`.

**Comment by zekefarwell** [comment]
Wow, that was fast @cmuench!  I tested 7.2.0-dev and it works as expected now.  Thank you!
